### ISSUE TITLE
Camera accessory doesn't load

### ISSUE THREAD

**Issue by ladvoc** [luisiam/homebridge-foscam2#3]
My configuration file:

``` json
{
    "bridge": {
        "name": "Raspberry Pi",
        "username": "CC:22:3D:E3:CE:30",
        "port": 51826,
        "pin": "321-43-519"
    },

    "description": "",

    "accessories": [],

    "platforms": [{
        "platform": "Foscam2",
        "cameras": [{
            "username": "[myusername]",
            "password": "[mypassword]",
            "host": "192.168.1.22",
            "port": "88",
            "stay": "0",
            "away": "3",
            "night": "0",
            "path": ""
        }]
    }]

}
```

After following the installation instructions exactly, deleting the `persist` and `accessories` directories within `~/.homebridge`,  and triple checking my camera configuration, when I run HomeBridge, it still says 0 accessories have been loaded:

```
[Thu Aug 18 2016 03:24:02 GMT+0000 (UTC)] Loading 1 platforms...
[Thu Aug 18 2016 03:24:02 GMT+0000 (UTC)] Initializing Foscam2 platform...
[Thu Aug 18 2016 03:24:02 GMT+0000 (UTC)] Loading 0 accessories..
```

Then after it says `Homebridge is running on port 51826`, I receive the following errors:

```
[Thu Aug 18 2016 03:38:33 GMT+0000 (UTC)] [TypeError: Not a string or buffer]
[Thu Aug 18 2016 03:38:33 GMT+0000 (UTC)] Failed to detect API version!
```

My Foscam is a model FI9821P V2 and it worked fine with [homebridge-foscam](https://github.com/rooi/homebridge-foscam).

**Comment by luisiam** [comment]
Do you have the full log?

**Comment by ladvoc** [comment]
```
*** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
*** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
[Thu Aug 18 2016 14:29:50 GMT+0000 (UTC)] Loaded plugin: homebridge-foscam2
[Thu Aug 18 2016 14:29:50 GMT+0000 (UTC)] Registering platform 'homebridge-foscam2.Foscam2'
[Thu Aug 18 2016 14:29:50 GMT+0000 (UTC)] ---
[Thu Aug 18 2016 14:29:50 GMT+0000 (UTC)] Loaded config.json with 0 accessories and 1 platforms.
[Thu Aug 18 2016 14:29:50 GMT+0000 (UTC)] ---
[Thu Aug 18 2016 14:29:50 GMT+0000 (UTC)] Loading 1 platforms...
[Thu Aug 18 2016 14:29:50 GMT+0000 (UTC)] Initializing Foscam2 platform...
[Thu Aug 18 2016 14:29:50 GMT+0000 (UTC)] Loading 0 accessories...
Scan this code with your HomeKit App on your iOS device to pair with Homebridge:

    ┌────────────┐     
    │ 213-73-549 │     
    └────────────┘     

[Thu Aug 18 2016 14:29:50 GMT+0000 (UTC)] Homebridge is running on port 51826.
[Thu Aug 18 2016 14:29:52 GMT+0000 (UTC)] [TypeError: Not a string or buffer]
[Thu Aug 18 2016 14:29:52 GMT+0000 (UTC)] Failed to detect API version!
```

I should also note that HomeBridge pairs with my iOS device without any issues. The problem is that my device reports that the HomeBridge hub has no accessories:

![img_2524](https://cloud.githubusercontent.com/assets/3182119/17777886/f652342e-6516-11e6-9b0d-0dd38de8f6d4.PNG)

**Comment by luisiam** [comment]
Are you running it on a Raspberry Pi? Can you add `self.log(error);` after https://github.com/luisiam/homebridge-foscam2/blob/master/index.js#L107 and check if there's any extra logging? Try to run homebridge in debug mode to see if it's giving you more info. Thanks.

**Comment by ladvoc** [comment]
I've added `self.log(error);` after line 107 and put Homebridge in debug mode. There was nothing new in the log. To make sure I was editing the correct file, I added a `console.log("Hello world");` at the top of `index.js` which was logged when I ran Homebridge.

**Comment by luisiam** [comment]
I am sorry I don't have much time for this project right now. I have never seen this error before. I suspect that the error happens in the `configureCamera` method.
https://github.com/luisiam/homebridge-foscam2/blob/master/index.js#L133

If you can add some debug code and let me know where inside the method is throwing the error, it will help me a lot on debugging. Thanks.

**Comment by toshibochan** [comment]
Try to not change
        "stay": "13",
        "away": "15",
        "night": "14",
        "path": ""

**Comment by toshibochan** [comment]
"platforms": [{
    "platform": "Foscam2",
    "cameras": [{
        "username": "admin",
        "password": "password",
        "host": "192.168.1.10",
        "port": "88",
        "stay": "13",
        "away": "15",
        "night": "14

**Comment by ladvoc** [comment]
That didn't seem to help. I will do some debugging and share my results.

**Comment by toshibochan** [comment]
What camera you using?

**Comment by ladvoc** [comment]
FI9821P V2

**Comment by jacobgelman** [comment]
The problem seems to be caused by the `detectAPI` method of `FoscamPlatform`.
In the catch block of this method, `[TypeError: Not a string or buffer]` and `Failed to detect API version!` are both logged. I will continue debugging within this method. I also noticed that when my Foscam is unplugged, these errors are not printed.

**Comment by ladvoc** [comment]
I found the issue. When `getDevInfo()` is called in the `FoscamPlatform.getInfo` method, the MAC address in the returned info object is `Infinity`. When I tried hardcoding in the camera's MAC address (`info.mac = "[My Foscam's MAC address]"`) the two errors disappeared and everything started functioning properly. The only question is why the MAC address is `Infinity`. It seems to me there are two possibilities: there is an issue with my Foscam or network configuration or there is a bug in the `foscam-client` package. I will do some further debugging to find out which it is.

**Comment by luisiam** [comment]
I noticed that your camera is a POE version. I have been using the Wi-Fi version and that function may be returning the Wi-Fi MAC address only. That's probably why your camera returns `Infinity`. I do not find any reference on how to get the LAN MAC address in the CGI manual. If you can find anything, I can definitely implement some algorithm to prevent that. I don't have a POE version and I cannot test it out.

**Comment by ladvoc** [comment]
My Foscam is also equipped with Wi-Fi and that is how I have it connected. I tried the `getDevInfo` command using the Foscam's CGI and the proper MAC address was returned in the XML. I then tried running `foscam-client`'s `getDevInfo()` method again and found that `Infinity` was still returned. To me it appears the bug is a result of `foscam-client` parsing the response XML incorrectly. Should I close this issue and open one on `foscam-client` instead?
